src/data/multiset/basic.lean

```diff
@@ -2152,6 +2160,20 @@ begin
     simp only [count_repeat, eq_self_iff_true, if_true, card_repeat]},
 end
 
+lemma filter_eq (s : multiset α) (b : α) : s.filter (eq b) = repeat b (count b s) :=
```

Comment:
Can you add a variant for `(= b)` too? Perhaps:
```
lemma filter_eq' (s : multiset α) (b : α) : s.filter (= b) = repeat b (count b s) :=
begin
  ext a,
  rw [count_repeat, count_filter],
  simp_rw [←dif_eq_if],
  congr' 1 with rfl,
  refl,
end

lemma filter_eq (s : multiset α) (b : α) : s.filter (eq b) = repeat b (count b s) :=
by { simp_rw [←filter_eq', eq_comm], congr }
```
which also saves needing `eq_comm` in both proofs.

Reply:
Thanks for the suggestion, it helped me find an even shorter proof with `if_ctx_congr`